### PR TITLE
Added surveyAnsweredDb model to calc the answer ratio only if changes

### LINKED ISSUE #1534: Save answered ratio in db

### PR DESCRIPTION
**Pull request by idelcano**
Closes #1534

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyAnsweredRatioDB.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    /**
+     * Total number of compulsoryAnswered questions
+     */
+    @Column
+    int answered_compulsory_questions;
+
+    public SurveyAnsweredRatioDB() {
+    }
+
+    public static void saveEntityToModel(SurveyAnsweredRatio surveyAnsweredRatio){
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyAnsweredRatioDB.java]: SurveyAnsweredRatioDB should only have db data, to save SurveyAnsweredRatio we should create a repository

File: app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyAnsweredRatioDB.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+            surveyAnsweredRatioDB = new SurveyAnsweredRatioDB();
+        }
+        surveyAnsweredRatioDB.id_survey = surveyAnsweredRatio.getSurveyId();
+        surveyAnsweredRatioDB.total_questions = surveyAnsweredRatio.getTotal();
+        surveyAnsweredRatioDB.answered_questions = surveyAnsweredRatio.getAnswered();
+        surveyAnsweredRatioDB.total_compulsory_questions = surveyAnsweredRatio.getTotalCompulsory();
+        surveyAnsweredRatioDB.answered_compulsory_questions = surveyAnsweredRatio.getCompulsoryAnswered();
+        surveyAnsweredRatioDB.save();
+    }
+    public static SurveyAnsweredRatioDB getSurveyAnsweredRatioBySurveyId(long id_survey){
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyAnsweredRatioDB.java]: SurveyAnsweredRatioDB should only have db data, to get SurveyAnsweredRatio we should create a repository

File: app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyDB.java [hunk 3]
```diff
@@ -786,6 +788,13 @@ public void reschedule(Date newScheduledDate, String comment) {
         this.save();
     }
 
+    public SurveyAnsweredRatioDB getSurveyAnsweredRatio() {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/data/database/model/SurveyDB.java]: to get SurveyAnsweredRatio we should not use the survey model but GetSurveyAnsweredRatioUseCase

File: app/src/main/java/org/eyeseetea/malariacare/domain/entity/Survey.java [hunk 4]
```diff
@@ -36,6 +38,9 @@ public void setStatus(int status) {
     }
 
     public SurveyAnsweredRatio getSurveyAnsweredRatio() {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/entity/Survey.java]: to get SurveyAnsweredRatio we should not use the survey model but GetSurveyAnsweredRatioUseCase

File: app/src/main/java/org/eyeseetea/malariacare/domain/entity/SurveyAnsweredRatio.java [hunk 5]
```diff
@@ -19,12 +19,18 @@
 
 package org.eyeseetea.malariacare.domain.entity;
 
+import org.eyeseetea.malariacare.data.database.model.SurveyAnsweredRatioDB;
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/entity/SurveyAnsweredRatio.java]: domain entity should have not dependency to data layer

File: app/src/main/java/org/eyeseetea/malariacare/domain/entity/SurveyAnsweredRatio.java [hunk 6]
```diff
@@ -141,4 +154,12 @@ public boolean isCompulsoryCompleted(){
         return compulsoryAnswered>=totalCompulsory;
     }
 
+    public static SurveyAnsweredRatio getModelToEntity(SurveyAnsweredRatioDB surveyAnsweredRatioDB){
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/entity/SurveyAnsweredRatio.java]: domain entity should not convert from model to entity, this a repository responsibility

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetSurveyAnsweredRatioUseCase.java [hunk 7]
```diff
@@ -3,17 +3,17 @@
...
-import org.eyeseetea.malariacare.domain.entity.SurveyAnsweredRatioCache;
 
 public class GetSurveyAnsweredRatioUseCase {
     public interface Callback{
         void nextProgressMessage();
         void onComplete(SurveyAnsweredRatio surveyAnsweredRatio);
     }
 
-    public enum RecoveryFrom {DATABASE, MEMORY_FIRST }
+    public enum Action {FORCE_UPDATE, GET}
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/GetSurveyAnsweredRatioUseCase.java]: We should not couple get and save AnsweredRatio processes, this use case should only get from database and we should create a SaveAnsweredRatioUseCase

File: app/src/main/java/org/eyeseetea/malariacare/layout/adapters/dashboard/ADashboardAdapter.java [hunk 8]
```diff
@@ -201,7 +200,10 @@ public void onComplete(SurveyAnsweredRatio surveyAnsweredRatioResult) {
                         Log.d(getClass().getName(), "onComplete");
                     }
                 });
-        SurveyAnsweredRatio surveyAnsweredRatio = SurveyAnsweredRatioCache.get(survey.getId_survey());
+        SurveyAnsweredRatio surveyAnsweredRatio = SurveyAnsweredRatio.getModelToEntity(survey.getSurveyAnsweredRatio());
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/layout/adapters/dashboard/ADashboardAdapter.java]: We should use GetAnsweredRatioUseCase here

**Review by xurxodev — CHANGES_REQUESTED** [review]
@idelcano I have requested you some changes, can you review it please?

**Comment by manuelplazaspalacio** [comment]
Changes done @xurxodev

**Review by xurxodev — APPROVED** [review]
thanks @idelcano and @manuelplazaspalacio

**Pull request merged**